containers/org.eclipse.linuxtools.docker.ui/src/org/eclipse/linuxtools/internal/docker/ui/jobs/CopyFromDockerJob.java

```diff
@@ -0,0 +1,1233 @@
+	private final Path m_targetfolder;
+	/** Whether this is a mirror-job */
+	private final boolean m_mirror;
+	/** The paths to copy */
+	private Map<Path, Path> m_pathsToCopy;
+	/** The container to copy from */
+	private String m_containerId = null;
+	/** The id of the image-id of the image */
+	private String m_imageId = null;
+
+	/**
+	 * Get the global set of copied paths and copy it into this job
+	 *
+	 * @param folder
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	public static Set<IPath> getCopyiedPaths(Path folder) {
+
+		HashSet<Path> ps = getCopiedSet(folder);
+		synchronized (m_lockObject) {
+			return ((Set<IPath>) ps.clone());
+		}
+	}
+
+	/**
+	 * The type of copy operation
+	 */
+	public enum CopyType {
+		/** Normal copy from an image */
+		Image,
+		/** Normal copy from a container */
+		Container,
+		/** Mirror (follow symlinks) from an image */
+		ImageMirror,
+		/** Mirror (follow symlinks) from a container */
+		ContainerMirror
+	}
+
+	/**
+	 * Copy a path from a docker image or container to the local system If copying
+	 * from an image, the container will be started. Container and Image must
+	 * provide the <code>ls</code> command within the <code>PATH</code> Images must
+	 * also provide the <code>sleep</code> command within the <code>PATH</code>
+	 *
+	 * @param connection The docker connection to use
+	 * @param copyType   The kind of copy operation
+	 * @param desc       The name or hash of the image or container.
+	 * @param copySet    The paths to copy to the target folder
+	 * @param targetDir  Where to copy folder
+	 */
+	public CopyFromDockerJob(IDockerConnection connection, CopyType copyType, String desc, Set<Path> copySet,
+			Path targetDir) {
+		this(connection, copyType, desc, copySet.stream().collect(Collectors.toMap(x -> x, x -> targetDir)));
+	}
+
+	/**
+	 * Copy a path from a docker image or container to the local system If copying
+	 * from an image, the container will be started. Container and Image must
+	 * provide the <code>ls</code> command within the <code>PATH</code> Images must
+	 * also provide the <code>sleep</code> command within the <code>PATH</code>
+	 *
+	 * @param connection The docker connection to use
+	 * @param copyType   The kind of copy operation
+	 * @param desc       The name or hash of the image or container.
+	 * @param copyMap    A map of source and destinations folders. In mirror-mode
+	 *                   they must all point to the same target folder.
+	 */
+	public CopyFromDockerJob(IDockerConnection connection, CopyType copyType, String desc, Map<Path, Path> copyMap) {
+
+		// TODO set title based on ct
+		super(Messages.getString(COPY_VOLUMES_FROM_JOB_TITLE));
+
+
+		m_connection = (DockerConnection) connection;
+
+
+		m_containerId = null;
+		if (copyType == CopyType.Container || copyType == CopyType.ContainerMirror) {
+			m_image = null;
+			m_containerId = desc;
+		} else if (copyType == CopyType.ImageMirror || copyType == CopyType.ContainerMirror) {

```

Comment:
I believe this line should be copyType == CopyType.Image || copytType == CopyType.ImageMirror